.goreleaser.yml

```diff
@@ -1,47 +1,75 @@
 archives:
   - files:
-      - none*
+      # Ensure only built binary is archived
+      - 'none*'
     format: zip
     name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
 before:
   hooks:
-    - go mod download
+    - 'go mod download'
 builds:
-  - binary: '{{ .ProjectName }}_{{ .Version }}'
+  - # Binary naming only required for Terraform CLI 0.12
+    binary: '{{ .ProjectName }}_v{{ .Version }}_x5'
+    env:
+      - CGO_ENABLED=0
     flags:
       - -trimpath
-    goarch:
-      - '386'
-      - amd64
-      - arm
-      - arm64
     goos:
       - darwin
       - freebsd
       - linux
       - windows
+    goarch:
+      - '386'
+      - amd64
+      - arm
+      - arm64
     ignore:
-      - goarch: '386'
-        goos: darwin
+      - goarch: arm
+        goos: windows
+      - goarch: arm64
+        goos: freebsd
+      - goarch: arm64
+        goos: windows
     ldflags:
-      - -s -w -X version.ProviderVersion={{.Version}}
+      - -s -w -X main.Version={{.Version}}
```

Comment:
We define `ProviderVersion` in `version/version.go` and reference it when building the HTTP User Agent string for AWS API calls (and that version is I believed used in some metrics somewhere).
If the new standard is `main.Version` (which makes sense to me) then let's move to that as part of this PR.

Reply:
keep it as `version.ProviderVersion` in [9e65395](https://github.com/hashicorp/terraform-provider-aws/pull/23102/commits/9e653954bf6341d9785a13fa7f373f608ae629b5)